site/docs/v1/tech/guides/user-search-with-elasticsearch.adoc

```diff
@@ -0,0 +1,333 @@
+
+| `query`
+| Yes
+| Useful to leverage the full power of Elasticsearch queries and need to look across different fields, including nested values, and/or use compound or complex search parameters such as ranges.
+
+|===
+
+
+==== Ids Searches
+
+`ids` searches are useful when you know exactly which users you want to retrieve. This is the only search which is guaranteed to only query the database, never Elasticsearch.
+
+==== QueryString Searches
+
+The `queryString` search is a case-insensitive match. Whitespace is allowed in the search, but must be URL escaped; for example, using `%20` for a space character. Elasticsearch compatible regular expressions may be used, so you may search by prefix or suffix using the `*` wildcard. You may also search particular fields by prefixing the query with a field name, such as `email:`.
+
+If the `queryString` parameter has a `:` it will be passed as-is. Otherwise, if it has no spaces and contains a `@` it is to be an email address and will be passed with a prefix: `email:`. Otherwise, if it has no spaces it will have the wildcard `*` prepended and appended.
+
+==== Query Searches
+
+The `query` parameter search requires an escaped JSON string which is passed to Elasticsearch and therefore must be a valid Elasticsearch query. With a `query` search, you have the full power of the Elasticsearch query language.
+
+=== Field Mappings
+
+For both `queryString` and `query` searches, you may search against specific fields.
+
+The exact list of indexed fields is not documented, but if you are running FusionAuth self-hosted, you can find the list by https://www.elastic.co/guide/en/elasticsearch/reference/6.3/indices-get-mapping.html[retrieving the Elasticsearch mapping].
+
+If you are using FusionAuth Cloud, please https://account.fusionauth.io/account/support[open a support ticket] if you have a question about which fields are searchable.
+
+== Results Of a User Search
+
+The results of any search have the same format. There is a `total` object with the number of results and a `users` array containing user objects.
+
+[source,json] 
+.Sample Results From a Search
+----
+include::docs/src/json/users/users-search-response.json[]
+----
+
+== User Search Examples
+
+Below are examples of searches you can run. All examples are available to be run locally by downloading https://github.com/fusionauth/fusionauth-example-user-search[this GitHub repository] and following the instructions in the README.
+
+Each example uses the below shell script to run the search. Feel free to download it, update your API key and base FusionAuth URL, and experiment. The script uses the `POST` method so that the request body can be large. The value of the filename passed to the script changes each time, but everything else is the same. 
+
+[source,shell] 
+.`search.sh`: An Example Shell Script For Searching
+----
+include::https://raw.githubusercontent.com/FusionAuth/fusionauth-example-user-search/main/examples/search.sh[]
```

Comment:
When this is rendered, the #!/bin/sh is hardly visible. Is that easily changed?

Reply:
Only took a major version upgrade of the build tool :)